dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.test.tsx

```diff
@@ -159,4 +160,34 @@ context("when there is a valid resouce", () => {
     );
     expect(wrapper.find(OtherResourceItem)).toExist();
   });
+
+  it("ignores the namespace if it matches a known type", () => {
+    const deployment = {
+      metadata: {
+        name: "foo",
+        selfLink: "/api/v1/namespaces/deployments/secrets/kubernetes",
+      },
+      status: { replicas: 1, updatedReplicas: 1, availableReplicas: 1 },
+    } as IResource;
+    kubeItem.item = deployment;
+    const wrapper = shallow(
+      <ResourceTableItem {...defaultProps} resource={kubeItem} name={deployment.metadata.name} />,
+    );
+    expect(wrapper.find(SecretItem)).toExist();
+  });
+
+  it("ignores the resource name if it matches a known type", () => {
```

Comment:
I think what you mean in both cases is something like: "it parses the type from the position in the self-link". I was confused at first as it's not that anything is being ignored, but rather, prior to this change, the position was not being used and so any matching type string would match?

Reply:
Yes, maybe I didn't explained myself. I wanted to say that if a secret is deployed in the namespace `services` or is called `services` it still gets detected as a secret